userland/userland_generic.ld

```diff
@@ -20,6 +20,14 @@ MEMORY {
 }
 
 SECTIONS {
+    /* App state section. Used for persistent app data.
+     * Goes before the Text section in Flash.
+     */
+    .app_state :
+    {
+        KEEP (*(.app_state))
+    } > FLASH =0xFF
+
```

Comment:
This change shouldn't matter in practice. It's up to elf2tbf to decide what order things are put in.

Reply:
Unfortunately, that's not correct. If you look at https://github.com/helena-project/tock/blob/b0579e6d710e9b390220a2d7577215684a050a3d/userland/userland_generic.ld#L52 you'll see that the application header is specified as offsets from the beginning of text in the LD file. With .app_state where it was, the GOT offset was off by 1024 Bytes.

Reply:
Ah good point. I know we discussed this before but it's weird for the linker to be doing this work since elf2tbf has to follow its lead, rather than just requiring elf2tbf to set those values. Seems like this kind of thing might keep happening.